part1-장재혁-week4/src/signin.js

```diff
@@ -0,0 +1,77 @@
+
+const emailInput = document.getElementById('emailInput');
+const passwordInput = document.getElementById('passwordInput');
+const ctaSignIn = document.getElementById('ctaSignIn');
+let errorEmail = document.getElementById('errorEmail');
+let errorPassword = document.getElementById('errorPassword');
+
+let eyeButtons = document.getElementsByClassName('eye-button');
+let eyeOffs = document.getElementsByClassName('eyeOff');
+
+
+const emailCheck = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+emailValidChecker =(email) => {
+    if(emailCheck.test(email) === false) return false;
+    else return true;
+}
+
+emailInput.addEventListener('focusout', () => {
+    if (emailInput.value === '') {
+        errorEmail.textContent = '이메일을 입력해 주세요.';
+        emailInput.classList.add('invalid'); 
+    } else if (!(emailValidChecker(emailInput.value))){
+        errorEmail.textContent = '올바른 이메일 주소가 아닙니다.'; 
+        emailInput.classList.add('invalid'); 
+    } else {
+        errorEmail.textContent = '';
+        emailInput.classList.remove('invalid'); 
+    }

```

Comment:
early return 적용하시는 것이 성능상으로도 가독성 측면에서도 좋을 거 같아요!
```suggestion
    if (emailInput.value === '') {
        errorEmail.textContent = '이메일을 입력해 주세요.';
        emailInput.classList.add('invalid'); 
        return;
    } 
    
    if (!(emailValidChecker(emailInput.value))){
        errorEmail.textContent = '올바른 이메일 주소가 아닙니다.'; 
        emailInput.classList.add('invalid'); 
        return;
    } 
        errorEmail.textContent = '';
        emailInput.classList.remove('invalid'); 
    }

```